From 5bda80248062896ffa254f3d81217838ef1019cb Mon Sep 17 00:00:00 2001
From: Carlos Maiolino <cmaiolino@redhat.com>
Date: Wed, 14 Oct 2015 11:00:07 +0200
Subject: [PATCH] Fix signal error during backport

Backport of fuse patch bf5ffb5fd8558bd799791834def431c0cee5a11f introduced a
signal error while checking for errors in mountpoint_fd. This patch fixes the
signal.

Signed-off-by: Carlos Maiolino <cmaiolino@redhat.com>
---
 util/fusermount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/fusermount.c b/util/fusermount.c
index 9043008..e1b601b 100644
--- a/util/fusermount.c
+++ b/util/fusermount.c
@@ -1088,7 +1088,7 @@ static int mount_fuse(const char *mnt, const char *opts)
 	if (geteuid() == 0) {
 		res = add_mount(source, mnt, type, mnt_opts);
 		if (res == -1) {
-			if (isdir && mountpoint_fd != 1) {
+			if (isdir && mountpoint_fd != -1) {
 				res = fchdir(mountpoint_fd);
 				if (res == -1) {
 					close(mountpoint_fd);
-- 
2.4.3

